BraintreeCore/src/main/java/com/braintreepayments/api/BraintreeException.kt

```diff
@@ -0,0 +1,19 @@
+package com.braintreepayments.api
+
+import androidx.annotation.RestrictTo
+import java.io.IOException
+
+/**
+ * Parent class for exceptions encountered when using the SDK.
+ */
+open class BraintreeException : IOException {
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    constructor() : super()
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    constructor(message: String?) : super(message)
+
+    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+    constructor(message: String?, cause: Throwable?) : super(message, cause)
```

Comment:
maybe im missing a case, but could we do something like this: 
```kotlin
open class BraintreeException @JvmOverloads constructor(message: String? = null, cause: Throwable? = null): IOException(message, cause)
```
I dont know if the `@RestricTo` can go besides `@JvmOverloads` or at the top of the class. But ultimatley, we could do:

```kotlin
open class BraintreeException : IOException {

    @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
    @JvmOverloads 
    constructor(message: String? = null, cause: Throwable? = null) : super(message, cause)

}
```

Reply:
Nope I think that would work. This is a side effect of using the code conversion tool. `@JvmOverloads` is a good call.

Reply:
Honestly not sure why we have a constructor with no message lol. Like wtf